### ISSUE TITLE
“Unexpected end of JSON input ” 频繁出现这个问题

### ISSUE THREAD

**Issue by maoyuching** [lianginx/chatgpt-nuxt#12]
我：the map struct in javascript
chatgpt：In JavaScript, a "map" is a built-in data structure that allows you to store key-value pairs, where the keys can be of any type (including objects) and the values can be of any type (including other maps, arrays, or functions).

Here is an example

**Unexpected end of JSON input**

出现得挺频繁的，大佬有时间可以看一下吗

**Comment by Wenenenenen** [comment]
我也类似：Unterminated string in JSON at position 112

**Comment by itTkm** [comment]
I have investigated and fixed this issue: #16

## Investigation Result

I confirmed that this issue occurs when data received through the stream is incorrectly divided. This causes errors at `JSON.parse`, as the data is cut off in the middle of the JSON format.

The following example shows a case where the stream is split into two parts and it works correctly:

### Successful Case

The first stream:
```bash
data: {"id":"chatcmpl-6wbHdXzs8Cgh4arATBF53vGOV1YmB","object":"chat.completion.chunk","created":1679424361,"model":"gpt-3.5-turbo-0301","choices":[{"delta":{"role":"assistant"},"index":0,"finish_reason":null}]}

data: {"id":"chatcmpl-6wbHdXzs8Cgh4arATBF53vGOV1YmB","object":"chat.completion.chunk","created":1679424361,"model":"gpt-3.5-turbo-0301","choices":[{"delta":{"content":"H"},"index":0,"finish_reason":null}]}

data: {"id":"chatcmpl-6wbHdXzs8Cgh4arATBF53vGOV1YmB","object":"chat.completion.chunk","created":1679424361,"model":"gpt-3.5-turbo-0301","choices":[{"delta":{"content":"e"},"index":0,"finish_reason":null}]}

```

The second stream:
```bash
data: {"id":"chatcmpl-6wbHdXzs8Cgh4arATBF53vGOV1YmB","object":"chat.completion.chunk","created":1679424361,"model":"gpt-3.5-turbo-0301","choices":[{"delta":{"content":"l"},"index":0,"finish_reason":null}]}

data: {"id":"chatcmpl-6wbHdXzs8Cgh4arATBF53vGOV1YmB","object":"chat.completion.chunk","created":1679424361,"model":"gpt-3.5-turbo-0301","choices":[{"delta":{"content":"l"},"index":0,"finish_reason":null}]}

data: {"id":"chatcmpl-6wbHdXzs8Cgh4arATBF53vGOV1YmB","object":"chat.completion.chunk","created":1679424361,"model":"gpt-3.5-turbo-0301","choices":[{"delta":{"content":"o"},"index":0,"finish_reason":null}]}

data: [DONE]
```

### Error Case

The following example shows a case where JSON parsing fails and an error as the current issue occurs:

The first stream:
```bash
data: {"id":"chatcmpl-6wbHdXzs8Cgh4arATBF53vGOV1YmB","object":"chat.completion.chunk","created":1679424361,"model":"gpt-3.5-turbo-0301","choices":[{"delta":{"role":"assistant"},"index":0,"finish_reason":null}]}

data: {"id":"chatcmpl-6wbHdXzs8Cgh4arATBF53vGOV1YmB","object":"chat.completion.chunk","created":1679424361,"model":"gpt-3.5-turbo-0301","choices":[{"delta":{"content":"H"},"index":0,"finish_reason":null}]}

data: {"id":"chatcmpl-6wbHdXzs8Cgh4arATBF53vGOV1YmB","object":"chat.completion.chunk","created":1679424361,"model":"gpt-3.5-turbo-0301","choices":[{"delta":{"content":"e"},"index":0,"finish_reason":null}]}

data: {"id":"chatcmpl-6wbHdXzs8Cgh4arATBF53vGOV1YmB","obj
```

The second stream:
```bash
ect":"chat.completion.chunk","created":1679424361,"model":"gpt-3.5-turbo-0301","choices":[{"delta":{"content":"l"},"index":0,"finish_reason":null}]}

data: {"id":"chatcmpl-6wbHdXzs8Cgh4arATBF53vGOV1YmB","object":"chat.completion.chunk","created":1679424361,"model":"gpt-3.5-turbo-0301","choices":[{"delta":{"content":"l"},"index":0,"finish_reason":null}]}

data: {"id":"chatcmpl-6wbHdXzs8Cgh4arATBF53vGOV1YmB","object":"chat.completion.chunk","created":1679424361,"model":"gpt-3.5-turbo-0301","choices":[{"delta":{"content":"o"},"index":0,"finish_reason":null}]}

data: [DONE]
```

## Solution

I have fixed this issue by combining the received stream data and parsing only the remaining data in each loop.
